Retrieve the latest ES version

ezimuel opened the pull request:
This PR adds the `get_latest_version()` function to retrieve the latest Elasticsearch version automatically.

acsnyder reviewed (approved):
LGTM!  At least on my Mac

Do you want to have a fallback in case the API response fails?

xeraa commented:
I think the beta / RC releases of 9.0.0 will be picked up as the last version, which is probably not what we want?
I just pulled the JSON file and tried it with a slightly modified shell script.

```sh
#!/bin/bash

version_sort() {
    awk -F'.' '
    {
        printf("%d %d %d %s\n", $1, $2, $3, $0)
    }' | sort -n -k1,1 -k2,2 -k3,3 | awk '{print $4}'
}

awk -F'"' '/"version": *"/ {print $4}' stack.json | version_sort | tail -n 1
```

xeraa commented:
Yes, I also think that start-local should only pick stable releases. But if https://artifacts.elastic.co/releases/stack.json adds the beta and RC releases for 9.0.0 (which I assume it will), this PR will pick them up as the latest / highest version.

ezimuel commented:
@acsnyder I think the https://artifacts.elastic.co/releases/stack.json should only show stable releases of Elasticsearch, right?

xeraa commented:
The file already includes releases like `8.0.0-rc1` etc

ezimuel commented:
@xeraa I didn't notice it, thanks! In that case I think we should remove the `x.y.z-*` versions in the awk expression. I'll propose a change.

ezimuel commented:
@xeraa and @acsnyder I removed the `beta` and `candidate` releases and added also the `GA` versions (since `8.0.0 GA` is a stable release but it needs to be converted in `8.0.0` removing the GA suffix). Moreover, I added the test for checking that the latest version is not empty and it's a valid `major.minor.patch` format.
